queue_job/migrations/10.0.1.0.0/post-migration.py

```diff
@@ -15,3 +20,27 @@ def migrate(cr, version):
         UPDATE queue_job
         SET method_name = reverse(split_part(reverse(func_name), '.', 1))"""
     )
+    migrate_args_kwargs(cr)
+
+
+def migrate_args_kwargs(cr):
+    """Extract args and kwargs from v9 func column, which is the tuple
+    (func_name, args, kwargs) pickled in a binary string. Ignore done
+    jobs for performance reasons"""
+    cr.execute(
+        'select id, func from queue_job where state not in %s',
+        (tuple([DONE]),),
+    )
+    for _id, func in cr.fetchall():
+        try:
+            func_name, args, kwargs = Unpickler(StringIO(func)).load()
+        except Exception as e:
```

Comment:
Can we log here?

Reply:
Actually, this is quite serious as we've seen starting jobs with this generates gigabytes of log and session files once Odoo starts up so please set jobs to state 'failed' + active = False in addition to the logging.

Reply:
Oh there is no active column after 8.0 anymore. How do we encode that a job is cancelled as opposed to failed then?

Reply:
good question, I keep this open for now

Reply:
https://github.com/OCA/queue/issues/342